### PR TITLE
Added land analysis test scripts

### PR DESCRIPTION
**Pull request by apreucil**
Addition of the freanalysis_land folder containing land,py which inherits the base analysis class and includes a few test cases for creating figures relating to soil carbon content and soil moisture in a global stand-alone land model run. Note that this script is for testing only and should not be treated as final versions of the land analysis package. Additionally, a python program is created in tests (my_land_analysis.py) which tests the use of the new package.

### CODE REVIEW

File: freanalysis_aerosol/freanalysis_aerosol/__init__.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 @dataclass
 class Metadata:
     """Helper class that stores the metadata needed by the plugin."""
-    frequency: str = "monthly"
+    frequency: str = "monthly"object
```
Comment (ilaflott) [inline freanalysis_aerosol/freanalysis_aerosol/__init__.py:14]: here and L210- remove `object`

File: freanalysis_land/freanalysis_land/land.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import cartopy
+import cartopy.crs as ccrs
+import datetime
+import pandas as pd
+import re
+import xarray as xr
+
+
+class LandAnalysisScript(AnalysisScript):
+    """Abstract base class for analysis scripts.  User-defined analysis scripts
```
Comment (ilaflott) [inline freanalysis_land/freanalysis_land/land.py:14]: adjust description- `LandAnalysisScript` is no longer an abstract base-class, but one that inherits from it

File: freanalysis_land/freanalysis_land/land.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        self.description = "This is for analysis of land model (stand-alone)"
+        self.title = "Soil Carbon"
+
+    def requires(self):
+        """Provides metadata describing what is needed for this analysis to run.
+
+        Returns:
+            A json string describing the metadata.
+        """
+        raise NotImplementedError("you must override this function.")
```
Comment (ilaflott) [inline freanalysis_land/freanalysis_land/land.py:32]: if the `return` on L33 is what you want, remove this `NotImplementedError`

File: freanalysis_land/freanalysis_land/land.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        The output file is saved with the format '<variable>_global_map.png'.
+
+        Examples:
+        --------
+        global_map(my_dataset, 'mrso', plt_time=0, colormap='coolwarm', title='Global Temperature Map')
+        """
+        lon = dataset.lon
+        lat = dataset.lat
+
+        if plt_time == None: plt_time=len(dates)-1
```
Comment (ilaflott) [inline freanalysis_land/freanalysis_land/land.py:81]: if `type(plt_time)` is supposed to be `int`, then the logical condition you really want here is `if plt_time is None:`

File: freanalysis_land/freanalysis_land/land.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        matplotlib.figure.Figure
+            The figure object containing the generated time series plot.
+        
+        Notes:
+        ------
+        The function filters the dataset based on the provided variable range, longitude, and latitude bounds. It then 
+        calculates the monthly and annual means of the specified variable and plots the seasonal and annual means.
+        
+        '''
+        if var_range != None:
```
Comment (ilaflott) [inline freanalysis_land/freanalysis_land/land.py:160]: `var_range != None` should be `var_range is not None`

File: freanalysis_land/freanalysis_land/land.py [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+            data_filtered = dataset.where((dataset[var] > 0) &
+                                        (dataset.lat >= minlat) & (dataset.lon >= minlon) &
+                                        (dataset.lat <= maxlat) & (dataset.lon <= maxlon))
+        data_filtered['time'] = dates_period
+
+        data_df = pd.DataFrame(index = dates_period)
+        data_df['monthly_mean'] = data_filtered.resample(time='YE').mean(dim=['lat','lon'],skipna=True)[var].values
+        data_df['monthly_shift'] = data_df['monthly_mean'].shift(1)
+
+        if timerange != None:
```
Comment (ilaflott) [inline freanalysis_land/freanalysis_land/land.py:174]: see above

File: tests/my_land_analysis.py
Comment (ilaflott) [inline tests/my_land_analysis.py:1]: test scripts for `pytest` traditionally have a name like `tests/test_my_land_analysis.py`

File: tests/my_land_analysis.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+
+from freanalysis_land.land import LandAnalysisScript
+land = LandAnalysisScript()
+land.run_analysis("/work/a2p/lm4p2sc_catalog.json","/work/a2p/")
```
Comment (ilaflott) [inline tests/my_land_analysis.py:5]: take these lines and wrap them in a `def test_land_analysis_script():`. At the end of this function should be an `assert some_condition` that checks for success of `land.run_analysis`

**Review by ilaflott — CHANGES_REQUESTED** [review]
This is an awesome effort to bring land-focused analysis scripts to this repository! 

I am requesting minor changes- mostly ones that seek to create some uniformity and consistency with other tests etc in pythonic-`FRE` land.

File: freanalysis_land/freanalysis_land/land.py (hunk 3 above)
Comment (apreucil) [reply freanalysis_land/freanalysis_land/land.py:32]: I was not taking advantage of this function for this version of the analysis, so this function is never called. I could comment it out for this implemenatation, perhaps?

File: freanalysis_aerosol/freanalysis_aerosol/__init__.py (hunk 1 above)
Comment (apreucil) [reply freanalysis_aerosol/freanalysis_aerosol/__init__.py:14]: Will have to check with @menzel-gfdl on this change

**Review by ilaflott — APPROVED** [review]
thank you for the fixes and your commitment to good code! I'm down with this merge

**Pull request merged**